### PR TITLE
Allow setup keys to be provided in a file

### PR DESCRIPTION
**Pull request by moosetheory**
## Describe your changes
Adds a flag and a bit of logic to allow a setup key to be passed in using a file. The flag should be exclusive with the standard `--setup-key` flag.

## Issue ticket number and link
N/A

### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

#### Why?
I am attempting to set up netbird automatically in a Lix environment, but the secrets library I use requires that I write the secrets to files and ask programs to use those at runtime. I believe this change will also allow someone to use docker secrets to provide a setup key.

I am unsure what/if changes should be made to any documentation or to the README file.

(I apologize if I have missed anything, this is my first time contributing to an open source project.)

### CODE REVIEW

**Comment by moosetheory** [comment]
Looking at the output from the Windows testing, it seems to be in the server code, and I don't know how my tweaks could have caused it to fail, but if someone could point me in the correct direction for that error, I will take a look at trying to figure it out.

File: client/cmd/login.go [hunk 1]
```diff
@@ -39,6 +39,13 @@ var loginCmd = &cobra.Command{
 			ctx = context.WithValue(ctx, system.DeviceNameCtxKey, hostName)
 		}
 
+		if setupKeyPath != "" && setupKey == "" {
+			setupKey, err = getSetupKeyFromFile(setupKeyPath)
```
Comment (mlsmaycon) [inline client/cmd/login.go:43]: Better to use a new variable instead of overwriting the global used in the flags.

File: client/cmd/up.go [hunk 2]
```diff
@@ -73,6 +73,13 @@ func upFunc(cmd *cobra.Command, args []string) error {
 		ctx = context.WithValue(ctx, system.DeviceNameCtxKey, hostName)
 	}
 
+	if setupKeyPath != "" && setupKey == "" {
```
Comment (mlsmaycon) [inline client/cmd/up.go:76]: can you move this to a single function?

File: client/cmd/root.go [hunk 3]
```diff
@@ -246,6 +249,11 @@ var CLIBackOffSettings = &backoff.ExponentialBackOff{
 	Clock:               backoff.SystemClock,
 }
 
+func getSetupKeyFromFile(setupKeyPath string) (string, error) {
```
Comment (mlsmaycon) [inline client/cmd/root.go:252]: let's use strings.TrimSpace and strings.Trim(string(data),"\n") to avoid issues

**Review by mlsmaycon — CHANGES_REQUESTED** [review]
Thanks for the PR @moosetheory. I've left some comments to it.

**Review by mlsmaycon — APPROVED** [review]
Thanks for the contribution @moosetheory. We will take the requested changes internally.

**Comment by moosetheory** [comment]
Oh, thanks! I've been running into some test issues and have not been able to figure them out. I'm glad you are able to!

Thanks a ton!

**Comment by mlsmaycon** [comment]
no worries, here is the PR in case you want to have a look https://github.com/netbirdio/netbird/pull/2411

**Pull request merged**